### ISSUE TITLE
Range Around Transform Zoomable Map

### ISSUE THREAD

**Issue by ZottiGabriele** [mapbox/mapbox-unity-sdk#610]
In the new 1.40 version of the Unity SDK the Abstract Map component have the new "Range Around Transform" Tile provider, that provides tiles based on the transform of an object, in my case the camera, that moves with keyboard inputs.
Now the problem comes with zooming, since moving the "Target Transform" doesn't update the center coordinates of the map, when you call the UpdateMap method you will always zoom to the center coordinates, the ones you initialize the map with.
On the other end, leaving the camera still and updating only the coordinates breaks  completely the placements of the tiles.

Tl;dr: There should be a way to update center map coordinates based on the Target Transform position using the new Range Around Transform Tile Provider.
 
To recreate: 
1) Load the ZoomableMap example 
2) Set the Abstract Map component to use the new Range Around Transform Tile Provider 
3) Set the camera transform as the Target Transform 
4) Hit play, move the camera transform, then try to zoom

**Comment by atripathi-mb** [comment]
@ZottiGabriele Could you send me a code snippet of how you are `zooming` ? 
> Hit play, move the camera transform, then try to zoom

Do you use the `QuadtreeCameraMovement` script to do that ? 
You would need the updated zoom level in the `UpdateMap` call to zoom the map. 

> There should be a way to update center map coordinates based on the Target Transform position using the new Range Around Transform Tile Provider.

`RangeAroundTransformTileProvider` uses the position of the transform to calculate the center of the map,  if you physically move the camera, you have to use some metric to convert the camera movement into a zoom level change and use the `UpdateMap` method to change the map accordingly. 

Example - https://github.com/mapbox/mapbox-unity-sdk/blob/49caf22af8e653d51a0525062aad22dd8c8c1941/sdkproject/Assets/Mapbox/Examples/Scripts/QuadTreeCameraMovement.cs#L116 
This uses the mouse scroll as a metric to update zoom.

**Comment by ZottiGabriele** [comment]
@atripathi-mb I think i haven't explained well enough my problem. I don't want to change the zoom level based on the camera transform, I currently do that with a script based on the QuadTreeCameraMovement.cs that you just mentioned.
In my project the camera transform is the target for the RangeAroundTransformTileProvider, so that when I move the camera transform the map updates correctly repositioning the tiles, though it does not change the actual AbstractMap.CenterLatitudeLongitude value.

```C#
AbstractMap _abstractMap;

void ZoomMapUsingMouse(float zoomFactor)
    {
        var zoom = Mathf.Max(0.0f, Mathf.Min(_abstractMap.Zoom + zoomFactor * _zoomSpeed, maxZoom));
        _abstractMap.UpdateMap(_abstractMap.CenterLatitudeLongitude, zoom);
    }
```
So when i go to use the UpdateMap in the snippet the _abstractMap.CenterLatitudeLongitude it's still the one the map started with, hence it does not zoom at the center of the map based on the movement the target transform has made.
[Here it's a link to a video I made about the issue](https://streamable.com/8sdab)

**Comment by atripathi-mb** [comment]
@ZottiGabriele Ah, thanks for that video. I feel `CameraBoundsTileProvider` is best suited for your requirements. Any reason why you chose `RangeAroundTransformTileProvider`  instead of `CameraBoundsTileProvider` ?

Another alternative could be to use your camera script to update the center as you pan and use `RangeAroundCenterTileProvider`. 

`RangeAroundTransformTileProvider` is meant to load tiles around a transform and in general, tile providers do not update the map. Map gets updated and the tile provider loads tiles based on updated map parameters. 

Btw, you can get the center lat long from the transform by using 
` abstractMap.WorldToGeoPosition(transform.localPosition , _map.AbsoluteZoom)`

**Comment by atripathi-mb** [comment]
@ZottiGabriele Let us know if we can help you with something. Closing this for now.
